### ISSUE TITLE
Get the `Makefile` running

### ISSUE THREAD

**Issue by carlislerainey** [pos5737/hw01-taylor-chewning#3]
This was optional in the HW, but I want you to get the `Makefile` running, just so you can see it in action.

I also want you to make a couple of other small changes so that you can see the files within the project interacting a bit.

This should take about 10 minutes. 

Here's what I want you to do:

----

**Rename the file `interesting example project bar chart.R` to `interesting-example-project-bar-chart.R`.** In RStudio, in the lower-right panel, click the *Files* tab and click the *Rename* button. Replace the spaces with `-`. 

*Background*: Alternatively, you could use the underscore (`_`) or camel case (`interestingExampleProjectBarChart.R`). I use `-` in file names. See [these R scripts](https://github.com/carlislerainey/latent-dissent/tree/master/R) for some examples. Spaces are to Makefiles as garlic is to vampires--deadly. Makefiles are the most common source of problems when you have spaces in file names. If you don't use Makefiles, these spaces might cause only one or two issues in the next five years, but those issues are usually bad.

----

**Save your colorful bar plot.** After you create the plot in `interesting-example-project-bar-chart.R`, add 

```
# save plot 
ggsave("colorful-bar-plot.png", height = 6, width = 8)
```

so that the script becomes 

```
library(ggplot2)

# From on a categorical column variable
g <- ggplot(mpg, aes(manufacturer))
g + geom_bar(aes(fill=trans), width = .75) + 
  theme(axis.text.x = element_text(angle=65, vjust=0.6)) +
  labs(title="Categorywise Bar Chart", 
       subtitle="Manufacturer of vehicles", 
       caption="Source: Manufacturers from 'mpg' dataset")

# save plot 
ggsave("colorful-bar-plot.png", height = 6, width = 8)
```

Run the script. You should see the file `colorful-bar-plot.png` appear in the files tab (perhaps at the bottom of the list).

----

**Add the `colorful-bar-plot.png` to the LaTeX document.** Insert the figure using the command 

```
% add the colorful bar plot
\includegraphics[width = 4in]{colorful-bar-plot.png}
```

after the first `\includegraphics`.  `doc.tex` should now be

```
% doc.tex: a code to include a plot

\documentclass{article}
\usepackage{graphicx} % useful for including graphics

\begin{document}

You should see the figure below.

% add the plot
\includegraphics[width = 4in]{plot.png}

% add the colorful bar plot
\includegraphics[width = 4in]{colorful-bar-plot.png}

\end{document}
```

Click *Compile PDF*. You should see your colorful bar plot included in the pdf now.

----

**Create the `Makefile`.

1. In RStudio, click *File* > *New File* > *Text File*. The click *File* > *Save*. Save this empty file as `Makefile`. 
1. Copy-and-paste the entire `Makefile` below into your empty `Makefile`. Click *File* > *Save* (or just click the blue floppy disk). Make sure that the indents that you see below get into RStudio as a *tab*, not as spaces. If they paste as spaces, then you must delete the spaces and replace them with tabs. Save again if you made changes.
1. Click *Build* > *Configure Build Tools...*. Click *Okay*. Right-click anywhere in the RStudio window and select *Reload*.
1. In the upper-right panel of RStudio, select the *Build* tab. Click *More* > *Clean All*. This runs the commands you see at the bottom of the `Makefile`, which delete the pdf and figures. 
1. To recreate the figures and pdf, just click `Build`. This runs every command in the `Makefile` in the order the dependencies require (which is the order I wrote the `Makefile` below). It creates the two figures and the pdf without you needing to run the two R scripts and compile the pdf as three separate steps.
1. If you want to do it again, click *More* > *Clean and Rebuild*. This deletes and then recreates all your files.

```
# Makefile
# be careful to use tabs **not spaces**!

# define "all"" as the main pdf
all: doc.pdf

# STEP #1
# create basic .png plot
# depends on the R script
plot.png: create-plot.R
	# run R code
	Rscript create-plot.R
	# remove junk R plot, see https://github.com/tidyverse/ggplot2/issues/2787
	rm -f Rplots.pdf  
    
# STEP #2
# create colorful bar plot
# depends on the R script
colorful-bar-plot.png: interesting-example-project-bar-chart.R
	# run R code
	Rscript interesting-example-project-bar-chart.R
	# remove junk R plot, see https://github.com/tidyverse/ggplot2/issues/2787
	rm -f Rplots.pdf      

# STEP #3
# compile .tex file to .pdf
# depends on the tex file and the two figures
doc.pdf: doc.tex plot.png colorful-bar-plot.png
	# first, generate the pdf
	pdflatex doc.tex
	# then, remove the auxillary files latex generates
	rm -f *.bbl *.log *.synctex.gz *.aux *.out

# delete these files to clean the directory
clean:
	rm -f plot.png  # deletes basic plot
	rm -f interesting-example-project-bar-chart.png
	rm -f doc.pdf  # deletes pdf document
``` 

Then you'll have seen a `Makefile` in action!

**Comment by taylorkchewning** [comment]
I just went through the process of creating a Makefile, and I'm glad that I did because I understand what it does much better now. From my understanding, it takes all the pieces and parts that I created using my various scripts, and updates my final document, the PDF. Therefore, if I make a change in my data, I can rerun the scripts, create new versions of the charts, and ultimately get my final document to be reflective of all of these changes and new charts. This is really cool!
